### DISCUSSION TITLE
Release 0.7.7 (1.21.50.26)

### DISCUSSION THREAD

**Discussion by bryhaw** [Mojang/minecraft-editor#205] [Announcements]
_[Read the previous release notes here](https://github.com/Mojang/minecraft-editor/discussions/204)_  

---    

## Summary  

Hello and happy preview day 🥳! We are now on 0.7.7 (1.21.50.26), as of Wednesday, October 30th, 2024. 

As a reminder of Editor API support:   

- Editor Extensions [overview article](https://learn.microsoft.com/en-us/minecraft/creator/documents/editorextensionsintroduction) and [API reference documentation](https://aka.ms/EditorAPI) on learn.microsoft.com  
- [Starter Kit](https://github.com/Mojang/minecraft-editor-extension-starter-kit): Contains the compiler tools and various libraries required to get started making Editor Extensions  
- [Samples](https://github.com/Mojang/minecraft-editor-extension-samples) created by our team to jump-start your own Extension creations  

Additionally, if you're new to the Editor GitHub this week, we encourage you to join the Bedrock Add-Ons Discord [#Editor channel](https://discord.gg/TZnQKFwGpt)!  

---

## Grapple locomotion updates

We have made updates to Grapple to make locomotion around your world more easily. These updates include:
- Changing the grapple-to-selection hot-key from F to CTRL+SHIFT+F (G still works for general grapple)
- Added 'getRay()' to Cursor Service API
- Camera now rotates towards grapple target during flight and is less snappy
- Grapple now lets you fly 100 blocks towards the mouse pointer if there is no nearby block to grapple to

https://github.com/user-attachments/assets/b1e0f6e6-7a58-4658-af10-4ef7c9999cd5

---

## Widget Component Updates

We have made several updates to Widget and Widget Components making it easier to use copied blocks from your clipboard. These include:
- Updating ClipboardWidgetComponent to the client widget system to allow creators to be able to add clipboard items to widgets, and have them displayed as 'holograms'
- Added a flag to Widget and WidgetBaseComponent to allow the Y position to lock to the surface, based on the XZ position
- Added a flag to Widget to allow the widget to bind it's position to the 3D block cursor

https://github.com/user-attachments/assets/9e396732-a6b8-4de0-9411-a5e3c1abcb5b

---

## Middle Mouse Block Selection

We have updated Middle Mouse Block Selection to work across all block selection, making it easier to switch to nearby blocks. 

https://github.com/user-attachments/assets/ef77075a-b73e-4376-9086-ffea4510c51e

---

## Editor
- Fixed a bug where cylinder and cuboid brushes with a height of 1 were incorrectly sized. 
- Renamed 'height' to 'length' for pyramid brush settings to be more consistent when rotated. 
- Fixed deselect in Line Tool (this was actually a problem with Client Widgets not correctly tracking selected state in script)
- Fix incorrect key binding in Selection move up/down
- Fixed 'Next Navigation Position' keyboard shortcut to no longer conflict with 'Previous Navigation Position'. 
- Brush Painting tools only work in the starting dimension (generally Overworld) and do not work when changing dimensions
- Selection tool size input box is now enabled when block is selected 
- Fixed keyboard navigation for interactive tooltip links 
- Fixed padding for Selection tool panel items when the scrollbar is absent 
- Changed `IModalToolContainer.addTool` API function to take unique tool id as a required argument. Removed `inputContextId` and `inputContextLabel` from creation parameters.
- Removed `hide`, `show`, `dispose` functions from `IModalTool` API
- Renamed `anchor` in the `ClipboardWriteOptions` to `normalizedOrigin` for clarity
- Added a new `ClipboardWidgetComponent` to the client widget system to allow creators to be able to add clipboard items to widgets, and have them displayed as 'holograms'
- Added a flag to `Widget` and `WidgetBaseComponent` to allow locking the Y position to the surface, based on the XZ position
- Added a flag to `Widget` to tell the widget to bind it's position to that of the 3D block cursor
- Added quick action button bar to Selection (Deselect & Fill)
- Added Copy & Cut to main menu and action bar items
- Removed all of the `CursorAttachment` related types, properties and methods
- Added information tooltip links for core editor panes 
- Middle mouse block picking in block picker modal, 
- Grapple locomotion updates. 
    - Added 'getRay()' to Cursor Service API
    - Camera now rotates towards grapple target during flight and is less snappy
    - Grapple now lets the creator fly 100 blocks distance towards the mouse pointer if there is no nearby block to grapple to
    - Changed hotkey for grapple-to-selection from F to CTRL+SHIFT+F
- Fixed issue where camera flashes to the inside of the player mobs' head when grappling. 

---   

Thank you for being here! You're the best! ❤️

**Comment by AbbottMc** [comment]
Cool, awsome update
